adafruit_esp32spi/adafruit_esp32spi_wsgiserver.py

```diff
@@ -216,3 +217,11 @@ def _get_environ(self, client):
             env[key] = value
 
         return env
+
+    def check_remote_ip(self):
```

Comment:
if we move the remote ip fetching to a property on `Socket`, then we're still going to want to expose it some way to a server application. Luckily, the WSGI defines an environment request variable named `REMOTE_ADDR`. You could add that to the `get_environ` function here: https://github.com/adafruit/Adafruit_CircuitPython_ESP32SPI/blob/master/adafruit_esp32spi/adafruit_esp32spi_wsgiserver.py#L174-L179

it would look something like

`env['REMOTE_ADDR'] = self._client_sock.remote_ip`

The environment object gets passed to the application on every single incoming request, so all your request handlers would just automatically get the remote IP and can use it for processing if they need it.